### PR TITLE
Implementation task: NotificationManager vs NotificationManagerCompat

### PR DESCRIPTION
**Pull request by anikiki**
Task/Issue URL: https://app.asana.com/0/488551667048375/1203771987668114/f

### Description
Replaced `NotificationManager` with `NotificationManagerCompat`.

### Steps to test this PR

- [ ] Install from this branch.
- [ ] Smoke test VPN notifications.
- [ ] Optionally smoke test other notifications.

### CODE REVIEW

File: app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/ui/notification/DeviceShieldAlertNotificationBuilder.kt [hunk 1]
```diff
@@ -68,12 +69,11 @@ class AndroidDeviceShieldAlertNotificationBuilder constructor(
...
     private fun registerAlertChannel(context: Context) {
         if (appBuildConfig.sdkInt >= Build.VERSION_CODES.O) {
-            val notificationManager = context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+            val notificationManager = NotificationManagerCompat.from(context)
             if (notificationManager.getNotificationChannel(VPN_ALERTS_CHANNEL_ID) == null) {
-                val channel = NotificationChannel(VPN_ALERTS_CHANNEL_ID, VPN_ALERTS_CHANNEL_NAME, NotificationManager.IMPORTANCE_DEFAULT)
+                val channel = NotificationChannel(VPN_ALERTS_CHANNEL_ID, VPN_ALERTS_CHANNEL_NAME, IMPORTANCE_DEFAULT)
                 channel.description = VPN_ALERTS_CHANNEL_DESCRIPTION
                 notificationManager.createNotificationChannel(channel)
-                notificationManager.isNotificationPolicyAccessGranted
```
Comment (anikiki) [inline app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/ui/notification/DeviceShieldAlertNotificationBuilder.kt:76]: Removed this line. I might be wrong, just don't see how it is used.

File: app/src/androidTest/java/com/duckduckgo/app/notification/NotificationRegistrarTest.kt [hunk 2]
```diff
@@ -58,7 +55,6 @@ class NotificationRegistrarTest {
         testee = NotificationRegistrar(
             TestScope(),
             context,
-            notificationManager,
```
Comment (anikiki) [inline app/src/androidTest/java/com/duckduckgo/app/notification/NotificationRegistrarTest.kt:61]: Got rid of `NotificationManager` here and only use `NotificationManagerCompat`.

File: app/src/main/java/com/duckduckgo/app/di/NotificationModule.kt [hunk 3]
```diff
@@ -40,12 +39,6 @@ import dagger.SingleInstanceIn
 @Module(includes = [DaoModule::class])
 object NotificationModule {
 
-    @Provides
-    @SingleInstanceIn(AppScope::class)
-    fun provideNotificationManager(context: Context): NotificationManager {
```
Comment (anikiki) [inline app/src/main/java/com/duckduckgo/app/di/NotificationModule.kt:45]: No need anymore to provide the `NotificationManager`.

File: app/src/main/java/com/duckduckgo/app/notification/NotificationRegistrar.kt [hunk 4]
```diff
@@ -81,14 +79,10 @@ class NotificationRegistrar @Inject constructor(
...
-    @Suppress("NewApi") // we use appBuildConfig
+    @Suppress("NewApi") // we use NotificationCompatManager to retrieve channels
     override fun onResume(owner: LifecycleOwner) {
         val systemEnabled = compatManager.areNotificationsEnabled()
-        val allChannelsEnabled = when {
-            appBuildConfig.sdkInt >= O -> manager.notificationChannels.all { it.importance != IMPORTANCE_NONE }
-            else -> true
-        }
-
+        val allChannelsEnabled = compatManager.notificationChannels.all { it.importance != IMPORTANCE_NONE }
```
Comment (anikiki) [inline app/src/main/java/com/duckduckgo/app/notification/NotificationRegistrar.kt:85]: Removed the `when` as now we're using the `NotificationCompatManager` and `allChannelsEnabled` returns an empty collection for OS versions < O. This means `allChannelsEnabled=true` for OS versions < O.

**Review by malmstein — APPROVED** [review]
Forgot about this, sorry @anikiki !

**Pull request merged**